### DISCUSSION TITLE
nautilus 文件管理器无法通过“在终端打开”启动 deepin-terminal

### DISCUSSION THREAD

**Discussion by duanyao** [linuxdeepin/developer-center#3022] [Features Request & Ideas | 特性请求 & 头脑风暴]
版本：
deepin os: 20.1
deepin-terminal: 5.4.0.6-1
nautilus: 3.26.3.1-1 (来自 panda仓库)
nautilus-open-terminal：0.20-51  (来自 panda仓库)

旧版本的 deepin os 和 deepin-terminal 是可以由 nautilus 的“在终端打开”启动的，升级到 20.1 后不行了。

nautilus 是通过 nautilus-open-terminal 插件来启动终端的，它调用的是 /usr/bin/gnome-terminal 。我的机器没有装 gnome-terminal ，创建了 /usr/bin/gnome-terminal 符号链接，指向其 /usr/bin/deepin-terminal 。

升级到 deepin 20.1 后，点击nautilus 的“在终端打开”，nautilus 的命令行中报错：
```
未知的选项 'x, c'
```
nautilus-open-terminal 传递给终端的命令行是：
```
/usr/bin/gnome-terminal -x /bin/sh -c cd '/home/duanyao/' && exec $SHELL
```
看来是因为最近的 deepin-terminal 不再支持 -x 和 -c 选项了？我希望能改回来，与gnome-terminal 保持一定的兼容性 。

**Comment by wangliang-terminal** [comment]
对于命令行参数这块的兼容后面是会考虑的，感谢您的反馈！

**Comment by meiyixiang** [comment]
Hello, I have recorded this problem, and I hope I can quickly return to this original ability
